File: 2/generator/MatrixLowTriangleGenerator.cpp
#include "../matrix/MatrixDiagonal.h"
#include "MatrixLowTriangleGenerator.h"
#include "../matrix/MatrixLowTriangle.h"
#include <random>

Matrix *MatrixLowTriangleGenerator::generate(int size) {
    MatrixLowTriangle *matrix = new MatrixLowTriangle(size);
    std::uniform_real_distribution<double> distrib(-MAX_ABSOLUTE_MATRIX_VALUE, MAX_ABSOLUTE_MATRIX_VALUE);
    for (int i = 0; i < size; i++) {
        for (int j = 0; j <= i; j++) {
            matrix->put(i, j, distrib(randomGenerator));
        }
    }
    return matrix;
}

MatrixLowTriangleGenerator::MatrixLowTriangleGenerator(std::mt19937 &randomGenerator) : randomGenerator(
        randomGenerator) {}

File: 3/generator/MatrixLowTriangleGenerator.py
from generator import MatrixGenerator
import matrix
import random


class MatrixLowTriangleGenerator(MatrixGenerator):
    def generate(self, size):
        new_matrix = matrix.MatrixLowTriangle(size)
        for i in range(size):
            for j in range(i + 1):
                new_matrix.put(i, j, random.uniform(-self.MAX_ABSOLUTE_MATRIX_VALUE, self.MAX_ABSOLUTE_MATRIX_VALUE))
        return new_matrix
